ledger-service/http-json/src/main/scala/com/digitalasset/http/dbbackend/ContractDao.scala

```diff
@@ -52,8 +52,19 @@ object ContractDao {
     new ContractDao(Connection.connect(jdbcDriver, jdbcUrl, username, password)(cs))
   }
 
-  def initialize(implicit log: LogHandler, sjd: SupportedJdbcDriver): ConnectionIO[Unit] =
-    sjd.queries.dropAllTablesIfExist *> sjd.queries.initDatabase
+  def initialize(checkIfExists: Boolean)(implicit
```

Comment:
I don’t think this works properly. Afaict if I start up without schema creation which is what most users do atm simply because that’s the main thing you would do before, we don’t check anything. However, that’s the most important case to check. You want to blow up if someone starts on the wrong schema.

I think the logic should be something like the following:
1. If some specifies that they want to recreate the schema, always recreate. This was the previous behavior and can be useful if for whatever reason things broke.
2. When starting up the JSON API (currently only possible if you didn’t initialize, with your PR also possible after initialization), validate that you run against a schema that you understand. I don’t think we need a flag to turn this off. If the schema has changed, you need to recreate or bad things will happen.
3. With `createSchemaIfNeeded` or whatever that flag is called, create the schema if it has the wrong version or has no version.

Reply:
true, completely oversaw this flaw.